### PR TITLE
WIP Improved search ui

### PR DESCRIPTION
**Pull request by wheelsandcogs**

### CODE REVIEW

File: src/search/index.scss [hunk 1]
```diff
@@ -2,7 +2,7 @@
   margin-bottom: 0;
 
   input {
-    width: 100%;
+    width: 96%;
```
Comment (lennym) [inline src/search/index.scss]: This hardly seems fragile at all. 

How does this scale to narrow windows?

File: src/filter-summary/index.jsx [hunk 2]
```diff
@@ -1,31 +1,74 @@
...
       totalCount: total
-    }
+    },
+    filters
   }
 }) => ({
-  total,
-  filtered
+  total: parseInt(total),
+  filtered: parseInt(filtered),
```
Comment (lennym) [inline src/filter-summary/index.jsx]:
```suggestion
  filtered: parseInt(filtered, 10),
```

File: src/filter-summary/index.jsx [hunk 3]
```diff
@@ -1,31 +1,74 @@
...
       count: filtered,
       totalCount: total
-    }
+    },
+    filters
   }
 }) => ({
-  total,
-  filtered
+  total: parseInt(total),
```
Comment (lennym) [inline src/filter-summary/index.jsx]:
```suggestion
  total: parseInt(total, 10),
```

File: src/filter-summary/index.jsx [hunk 4]
```diff
@@ -1,31 +1,74 @@
...
+}) => {
+  const searchString = filters.active['*'] && filters.active['*'][0];
+  const activeFilters = omit(filters.active, '*');
+
+  const removeFilter = filterKey => {
+    return e => {
+      e.preventDefault();
+      const urlParams = qs.parse(window.location.search, { ignoreQueryPrefix: true });
+      delete urlParams['filters'][filterKey];
+      window.location.href = `//${window.location.host + window.location.pathname + qs.stringify(urlParams, { addQueryPrefix: true })}`;
```
Comment (lennym) [inline src/filter-summary/index.jsx:23]: If we know what the end state of the query string after clicking a link should be then can we pre-calculate and just hard-code the links?

Comment (lennym) [reply src/filter-summary/index.jsx:23]: Reason: be slightly less dependent on client-side script.

File: src/filter-summary/index.jsx [hunk 5]
```diff
@@ -1,31 +1,74 @@
...
-  <h2 className="filter-summary">
-    {
-      filtered !== total
-        ? filteredLabel
-        : allShowingLabel
-    }
-  </h2>
-);
+  filteredLabel = filtered === 1 ? `${filtered} result` : `${filtered} results`,
+  allShowingLabel = total === 1 ? `${total} result` : `${total} results`,
```
Comment (joefitter) [inline src/filter-summary/index.jsx:12]:
```js
function pluralify(num) {
  return `${num} ${num === 1 ? 'result' : 'results'}`
}
```
?

Comment (joefitter) [reply src/filter-summary/index.jsx:12]: also, this seems like the wrong place to perform this logic, as it is _only_ performed if these arguments are omitted

**Comment by joefitter** [comment]
@wheelsandcogs have you exported this module? I can't see it being added to index

**Comment by joefitter** [comment]
ah sorry, ignore that, I thought this was a new thing. It is not.

File: src/filter-summary/index.jsx (hunk 5 above)
Comment (wheelsandcogs) [reply src/filter-summary/index.jsx:12]: We can probably remove these as props as I don't think they're actually passed in anywhere?

Comment (joefitter) [reply src/filter-summary/index.jsx:12]: that sounds like a great plan

**Comment by wheelsandcogs** [comment]
Not happening.

**Pull request closed without merging**